cmd/utils_test.go

```diff
@@ -587,40 +591,278 @@ func TestMoveFailPermissions(t *testing.T) {
+	}
+	t.Log("Created targetFile: ", targetFile)
 
+	// copy the source directory into the target
+	// this should fail as the target can not be an existing directory
+	err = cmd.CopyDir(log, sourceDir, targetFile)
 	if err != nil {
-		if !strings.Contains(err.Error(), "stat "+nonExistentFile+": no such file or directory") {
-			t.Errorf("String \"stat "+nonExistentFile+": no such file or directory\" not found in output: '%v'", err.Error())
+		if !strings.Contains(err.Error(), "Target "+targetFile+" exists. It should only be the name of the target directory for the copy") {
+			t.Errorf("String \"Target "+targetFile+" exists. It should only be the name of the target directory for the copy\" not found in output: '%v'", err.Error())
 		}
 	} else {
 		t.Error("Expected an error to be returned from command, but error was nil")
 	}
+
+}
+
+func TestCopyDirPass(t *testing.T) {
+	// set up logging
+	var outBuffer bytes.Buffer
+	log := &cmd.LoggingConfig{}
+	log.InitLogging(&outBuffer, &outBuffer)
+
+	// use sandbox directories and cleanup
+	sandbox, cleanup := cmdtest.TestSetupWithSandbox(t, true)
+	defer cleanup()
+
+	// create the directories in the sandbox
+
+	// create a base directory
+	baseDir := sandbox.ProjectDir
+	t.Log("baseDir is: ", baseDir)
+
+	// create a source directory
+	sourceDir := filepath.Join(baseDir, "sourceDir")
+	err := os.MkdirAll(sourceDir, 0755)
+	if err != nil {
+		t.Fatal("Error creating sourceDir: ", err)
+	}
+	t.Log("Created sourceDir: ", sourceDir)
+
+	// Create test file1 <name.extension>
+	file1 := filepath.Join(sourceDir, "file1.test")
+	data := []byte("home: " + sourceDir + "\n" + "generated-by-tests: Yes" + "\n")
+	err = ioutil.WriteFile(file1, data, 0644)
+	if err != nil {
+		t.Fatal("Error writing file1: ", err)
+	}
+	t.Log("Created file1: ", file1)
+
+	// Create test file2 <.extension>
+	file2 := filepath.Join(sourceDir, ".file2")
+	data = []byte("home: " + sourceDir + "\n" + "generated-by-tests: Yes" + "\n")
+	err = ioutil.WriteFile(file2, data, 0644)
+	if err != nil {
+		t.Fatal("Error writing file2: ", err)
+	}
+	t.Log("Created file2: ", file2)
+
+	// create a source sub directory
+	sourceSubDir := filepath.Join(sourceDir, "sourceSubDir")
+	err = os.MkdirAll(sourceSubDir, 0755)
+	if err != nil {
+		t.Fatal("Error creating source sub dir: ", err)
+	}
+	t.Log("Created sourceSubDir: ", sourceSubDir)
+
+	// Create test file3 <name.extension>
+	file3 := filepath.Join(sourceSubDir, "file3.test")
+	data = []byte("home: " + sourceSubDir + "\n" + "generated-by-tests: Yes" + "\n")
+	err = ioutil.WriteFile(file3, data, 0644)
+	if err != nil {
+		t.Fatal("Error writing file3: ", err)
+	}
+	t.Log("Created file3: ", file3)
+
+	// Create test file4 <.extension>
+	file4 := filepath.Join(sourceSubDir, ".file4")
+	data = []byte("home: " + sourceSubDir + "\n" + "generated-by-tests: Yes" + "\n")
+	err = ioutil.WriteFile(file4, data, 0644)
+	if err != nil {
+		t.Fatal("Error writing file4: ", err)
```

Comment:
It would be interesting to convert these into table test, but not very important